### ISSUE TITLE
Background loading / data pre-fetching

### ISSUE THREAD

**Issue by scottstanie** [opera-adt/dolphin#6]
Currently the data IO might be the largest percentage of the total run time prior to unwrapping.

- We should use something like https://github.com/isce-framework/isce3/blob/develop/python/packages/isce3/io/background.py to pre-fetch data blocks

**Comment by scottstanie** [comment]
This low GPU utilization might mean that the pre-fetching could be a big speedup:

![Image](https://user-images.githubusercontent.com/8291800/216126194-ead45772-8ad8-468a-8388-52a9fe51c20c.png)

**Comment by scottstanie** [comment]
Ok in my quick attempt to implement it here https://github.com/opera-adt/dolphin/pull/25, it's already working pretty well.

Total time:
```
[02/02 07:49:06] [INFO _log.py] Total elapsed time for run : 6.57 minutes (394.25 seconds)
```

- PS creation went from 160 seconds -> 60 seconds per burst
- Each burst wrapped phase went from  too about 3 minutes


GPU usage (closer together blue spikes means less waiting around time by the GPU)

<img width="1230" alt="image" src="https://user-images.githubusercontent.com/8291800/216376501-7ab368c7-4164-45cc-8516-81336477d1b3.png">

And indeed now the wrapped burst estimation is taking

**Comment by scottstanie** [comment]
Closed by #25
